package/Dockerfile

```diff
@@ -7,16 +7,19 @@ RUN mkdir -p /image/etc/ssl/certs /image/run /image/var/run /image/tmp /image/li
 
 FROM scratch
 ARG VERSION="dev"
+ARG PROG="k3s"
+
 COPY --from=base /image /
 RUN mkdir -p /etc && \
     echo 'hosts: files dns' > /etc/nsswitch.conf && \
-    echo "PRETTY_NAME=\"K3s ${VERSION}\"" > /etc/os-release && \
+    echo "PRETTY_NAME=\"${PROG} ${VERSION}\"" > /etc/os-release && \
     chmod 1777 /tmp
 VOLUME /var/lib/kubelet
-VOLUME /var/lib/rancher/k3s
+VOLUME /var/lib/rancher/${PROG}
 VOLUME /var/lib/cni
 VOLUME /var/log
 ENV PATH="$PATH:/bin/aux"
-ENV CRI_CONFIG_FILE="/var/lib/rancher/k3s/agent/etc/crictl.yaml"
-ENTRYPOINT ["/bin/k3s"]
+ENV CRI_CONFIG_FILE="/var/lib/rancher/${PROG}/agent/etc/crictl.yaml"
+
+ENTRYPOINT ["/bin/sh", "-c", "exec ${PROG}"]
```

Comment:
Ah, I guess if we're going to do this we need to make this an ENV and not just an ARG as args don't persist past the build stage, and this is being used a variable. Inspecting the resulting images shows that `${PROG}` is coming out as an empty string.

```
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/bin/aux",
                "CRI_CONFIG_FILE=/var/lib/rancher/k3s/agent/etc/crictl.yaml"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "#(nop) ",
                "CMD [\"agent\"]"
            ],
            "Image": "sha256:ee4f4659b81936abcb9210fadc90dbcad183993c9c093a22f80364af62a7e2fb",
            "Volumes": {
                "/var/lib/cni": {},
                "/var/lib/kubelet": {},
                "/var/lib/rancher/k3s": {},
                "/var/log": {}
            },
            "WorkingDir": "",
            "Entrypoint": [
                "/bin/sh",
                "-c",
                "exec ${PROG}"
            ],
```